pysonnet/projects.py

```diff
@@ -406,6 +417,115 @@ def make_sonnet_file(self, file_path, clean=True):
             os.mkdir(subfolder)
         log.debug("geometry project saved")
 
+    def export_current_density(self, frequency=None, driving_port= None,
+                               bottom=None, left=None, right=None, top=None,
+                               measurement_complex= None,measurement_type= None,
+                               levels_stop=None, levels_start= None):
+
+        """
+        An xml file is generated which contains the parameters of the current density
+        information to be exported as a csv file. Then a system command is run to
+        initiate the export. Allowed keywords are listed below.
+
+        :keyword frequency: float value given in Hz
+        :keyword region_style: 'Whole' or 'Rect'
+        :keyword driving_port: port number you wish to drive
+        :keyword bottom: The lower bound of the region
+        :keyword top: The upper bound of the region
+        :keyword left: The left bound of the region
+        :keyword right: The right bound of the region
+        :keyword levels_start: The beginning metal layer to export
+        :keyword levels_stop: The ending metal layer to export
+        :keyword measurement_type: 'jxy', 'jx', 'jy'
+        :keyword measurement_complex: 'yes' or 'No'
+
+        """
+
+
+        if [left,right,top,bottom] == [None,None,None,None]:
+            region_style = 'Whole'
+        else:
+            region_style = 'Rect'
+
+        # measurement type
+        if measurement_type is None:
+            type_error = "Set measurement type as 'jxy', 'jx', or 'jy'"
+            raise ValueError(type_error)
+
+        # make default one port
+        # grab the other parameters from sonnet file
+        # kwarg should be 'which port am i driving'
+
+        # Port1 parameters
+        capacitance = self.port['capacitance']
```

Comment:
There should be more than one port so this should be self.ports[0]['capacitance'] for the capacitance of the first port. Need to also check that the ports have been defined

Reply:
I need to make a loop that sets the port voltage to 1 for whatever the user sets and all other port voltages should be zero. But I am unclear on how the elements are saved in each self.port[0] = capacitance and how to make my code set the driving port first then all other ports second

Reply:
Okay, make `self.ports = {}` by default. When a port gets added in `add_port()` set `self.ports[number] = {'capacitance': capacitance, ...}`.

Then you can can access the parameters for a particular port number.
```
drive_port_params = self.ports[drive_port_number]  # get the drive port parameters
for number, params in self.ports.items():  # loop over ports
    if number != drive_port_number:  # but not the drive port
        # Do something here
```